packages/playwright/src/matchers/toEqual.ts

```diff
@@ -59,6 +59,21 @@ export async function toEqual<T>(
   if (pass) {
     printedExpected = `Expected: not ${this.utils.printExpected(expected)}`;
     printedReceived = `Received: ${this.utils.printReceived(received)}`;
+  } else if (Array.isArray(expected) && Array.isArray(received)) {
+    const normalizedExpected = expected.map((exp, index) => {
+      const rec = received[index];
+      if (exp instanceof RegExp)
```

Comment:
nit: please use `isRexExp` from 'playwright-core/lib/utils'.

Reply:
@yury-s 
I've applied the suggested change. Thanks for the helpful feedback! :)